File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

//dfs where we track the largest value recorded so far, 15:18
class Solution {
    //how do you iterate the size of the treeNode
    int goodNodes = 0;


    public void recursive(TreeNode currentNode, int maxVal){
        if(currentNode == null) return;

        if(currentNode.val >= maxVal){ 
            maxVal = currentNode.val; 
            goodNodes++;
            }
        else{
            
            System.out.println(currentNode.val);
            }

        recursive(currentNode.left, maxVal);
        recursive(currentNode.right, maxVal);

    }



    public int goodNodes(TreeNode root) {
        recursive(root, -100);
        return goodNodes;
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# Array of rightmost nodes at each level
# BFS, attach rightmost node (last node added)
#
##
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        unsearched = deque()
        unsearched.append(root)
        if root == None:
            return result

        while(unsearched):
            levelSize = len(unsearched)
            for i in range(levelSize):
                currentNode = unsearched.popleft()
                if currentNode.left:
                    unsearched.append(currentNode.left)
                if currentNode.right:
                    unsearched.append(currentNode.right)
            result.append(currentNode.val)
            
        
        return result
        